gitcha.go

```diff
@@ -156,3 +157,30 @@ func FindFirstFile(path string, list []string) (SearchResult, error) {
 
 	return SearchResult{}, nil
 }
+
+// deref takes a filename and recurses through symlinks until it
+// reaches the original filemame. There's a hard imposed maximum
+// depth to prevent infinite recursion.
+func deref(filename string) (os.FileInfo, error) {
```

Comment:
Maybe we should use `filepath.EvalSymlinks` here instead: https://pkg.go.dev/path/filepath#EvalSymlinks

Reply:
Thanks for the suggestion. Naturally, I looked for this function for a long time and couldn't find it, just to now see it right there, staring at me in the face. :) Updating PR.